### DISCUSSION TITLE
Flaky/sporradic PrismaClientInitializationError failures when connecting to PlanetScale

### DISCUSSION THREAD

**Discussion by EvHaus** [planetscale/discussion#556] [Bug Reports] [answered]
I'm using Prisma and PlanetScale for my app. I have unit tests which run on CI that connect to PlanetScale for some basic integration tests. It all works great, except that I'm seeing roughly 1 out of 10 test cases will fail with:

```
 FAIL  pages/api/property/POST.test.ts > POST /api/property > should create a new property
PrismaClientInitializationError: 
Invalid `OrganizationFactory.create()` invocation in
/home/runner/work/myproject/pages/api/property/POST.test.ts:52:15

  49 });
  50 
  51 it('should create a new property', async () => {
→ 52 	const org = await OrganizationFactory.create(
Can't reach database server at `ojj81b9i387e.us-west-2.psdb.cloud`:`3306`

Please make sure your database server is running at `oqq81b9i387e.us-west-2.psdb.cloud`:`3306`.
 ❯ Cn.handleRequestError node_modules/.pnpm/@prisma+client@5.4.2_prisma@5.4.2/node_modules/@prisma/client/runtime/library.js:123:7090
 ❯ Cn.handleAndLogRequestError node_modules/.pnpm/@prisma+client@5.4.2_prisma@5.4.2/node_modules/@prisma/client/runtime/library.js:123:6206
 ❯ Cn.request node_modules/.pnpm/@prisma+client@5.4.2_prisma@5.4.2/node_modules/@prisma/client/runtime/library.js:123:5926
 ❯ l node_modules/.pnpm/@prisma+client@5.4.2_prisma@5.4.2/node_modules/@prisma/client/runtime/library.js:128:9968
 ❯ Object.create node_modules/.pnpm/prisma-factory@0.1.12_@prisma+client@5.4.2_typescript@5.2.2/node_modules/prisma-factory/dist/index.js:109:18
 ❯ pages/api/property/POST.test.ts:52:15
     50| 
     51|  it('should create a new property', async () => {
     52|   const org = await OrganizationFactory.create();
       |               ^
     53| 
```

Usually retrying the tests makes the problem go away but the flakiness is starting to become more and more frustrating to deal with.

Before I go with a blunt solution like adding a retry-mechanism for all DB requests -- I'm wondering if there's any other recommended solutions. Why would establishing a connection to PlanetScale not work 100% reliably?

My app isn't live for users yet, and I'm worried that if my unit tests are catching such flakiness, when I launch the product my users might run into similar random connection failures too. Any advice from the PlanetScale on how to debug this further?

**Comment by orware** [comment, marked as answer]
Hello @EvHaus,

Based on  the hostname shown in your error message, `ojj81b9i387e.us-west-2.psdb.cloud`, your database and the associated credentials are likely over a year old and this means your connections are currently arriving directly onto our older edge routing infrastructure, rather than our current one.

Please generate a new set of credentials within the PlanetScale UI at your earliest convenience, which should present you with credentials that use the `aws.connect.psdb.cloud` hostname, which has improved reliability amongst other improvements.

Additionally, the following tip that suggests adding the use of the `connect_timeout` argument to your Prisma connection should also prove to be helpful:
https://planetscale.com/docs/prisma/prisma-best-practices#prisma-error-p1001

**Reply by EvHaus** [reply]
Thank you @orware. It's been a few days since I switched to the `aws.connect.psdb.cloud` hostname version and so far there haven't been any connection failures. 🤞
